config/src/main/java/org/springframework/security/config/annotation/web/configurers/HttpBasicConfigurer.java

```diff
@@ -95,6 +95,17 @@
 	 * @see HttpSecurity#httpBasic()
 	 */
 	public HttpBasicConfigurer() throws Exception {
+		withDefaults();
+	}
+
+	/**
+	 * Sets up the {@link HttpBasicConfigurer} using the default values for all of
+	 * its attributes.
+	 *
+	 * @return {@link HttpBasicConfigurer} for additional customization
+	 * @throws Exception
+	 */
+	public HttpBasicConfigurer<B> withDefaults() throws Exception {
```

Comment:
I think I'd prefer to revert and potentially use a generic noop Consumer. The reason is that `withDefaults` is exposed for those not using lambda's and becomes confusing since the defaults are already applied by default. Another reason is that way we don't need a withDefaults for every configurer that we have.

I know we already talked about creating a method reference for users, but in hindsight I'd suggest we just require users to create an empty lambda. The JDK doesn't even provide one. We can always introduce a generic no op consumer later on.

Reply:
I agree.
I believe users could still do the following to keep the defaults.
```
http
    .httpBasic(httpBasicConfigurer -> {});
```